From 88a6201917cad43fd4efea0f1c34c891b70a7414 Mon Sep 17 00:00:00 2001
From: =?utf8?q?P=C3=A1draig=20Brady?= <P@draigBrady.com>
Date: Thu, 25 Oct 2012 10:59:23 +0100
Subject: [PATCH] build: ensure factor links the iconv library

* src/local.mk (src_factor_LDADD): Append $(LIBICONV).
* crg.mk (sc_check-I18N-AUTHORS): A new syntax check rule
to ensure we add LIBICONV where appropriate.
* THANKS.in: Add the reporter.
Reported by Christian Jullien
Syntax check suggested by Jim Meyering
---
 THANKS.in    |    1 +
 cfg.mk       |    8 ++++++++
 src/local.mk |    1 +
 3 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/THANKS.in b/THANKS.in
index 504daf3..016a41e 100644
--- a/THANKS.in
+++ b/THANKS.in
@@ -118,6 +118,7 @@ Chris Sylvain                       csylvain@umm.edu
 Chris Yeo                           cyeo@biking.org
 Christi Alice Scarborough           christi@chiark.greenend.org.uk
 Christian Harkort                   christian.harkort@web.de
+Christian Jullien                   eligis@orange.fr
 Christian Krackowizer               ckrackowiz@std.schuler-ag.com
 Christian Rose                      menthos@menthos.com
 Christian von Roques                roques@pond.sub.org
diff --git a/cfg.mk b/cfg.mk
index 43fb22f..40ece55 100644
--- a/cfg.mk
+++ b/cfg.mk
@@ -181,6 +181,14 @@ sc_check-AUTHORS: $(all_programs)
 	  && diff $(au_actual) $(au_dotdot) \
 	  && rm -f $(au_actual) $(au_dotdot)
 
+# Ensure programs with authors with non ASCII names link with LIBICONV
+sc_check-I18N-AUTHORS:
+	@(cd $(srcdir)/src &&                           \
+	  for i in $$(git grep -l -w proper_name_utf8 *.c|sed 's/\.c//'); do \
+	    grep -E "^src_$${i}_LDADD"' .?= .*\$$\(LIBICONV\)' local.mk > \
+	      /dev/null || { echo FAIL $$i; exit 1; };            \
+	  done)
+
 # Look for lines longer than 80 characters, except omit:
 # - program-generated long lines in diff headers,
 # - tests involving long checksum lines, and
diff --git a/src/local.mk b/src/local.mk
index 6a01ef1..f40f681 100644
--- a/src/local.mk
+++ b/src/local.mk
@@ -277,6 +277,7 @@ src_cat_LDADD += $(LIBICONV)
 src_cp_LDADD += $(LIBICONV)
 src_df_LDADD += $(LIBICONV)
 src_du_LDADD += $(LIBICONV)
+src_factor_LDADD += $(LIBICONV)
 src_getlimits_LDADD += $(LIBICONV)
 src_printf_LDADD += $(LIBICONV)
 src_ptx_LDADD += $(LIBICONV)
-- 
1.7.2.5

